### DISCUSSION TITLE
sass source maps in dev => broken for me with vite 5.3.5+

### DISCUSSION THREAD

**Discussion by jrochkind** [ElMassimo/vite_ruby#522] [Q&A]
I swear I used to have sass source maps in dev working, but they seem to have stopped, and i'm not sure why. 

(By not working, I mean chrome dev tools is not using them. I haven't yet managed to figure out how to debug further). 

I do have it enabled in my vite.config.ts:

```ts
export default defineConfig({
  // enable sass sourcemaps -- vite supports sass sourcemaps with dev server only
  css: {
    devSourcemap: true
  },
```

Does anyone else have sass source maps in dev working?

vite_ruby 3.9.1
vite_rails 3.0.19

vite@5.4.11
sass@1.82.0

**Comment by jrochkind** [comment]
*update*: OK, I just figured this broke for me when updating vite versions, and narrowed it down to working with vite 5.3.4, and broken with vite 5.3.5 and later 5.3.x. 

(Have not been able to test with vite 6.x, because my app doesn't actually work with vite 6, not sure why)

I am trying to figure out how to make a non-vite-ruby reproduction app to see if I can reproduce this problem without vite-ruby -- but the simple "starter" vite apps end up injecting all CSS in a <style> tag on page, so you NEVER see file and line numbers in that app anyway, with or without source maps!

I am not sure how to make the simplest possible vite repro app that can demonstrate working source maps.  But I'm not really a vite or JS setup expert -- that's why I love vite-ruby, I normally don't have to be. 

@ElMassimo do you have any advice, either for creating a pure-vite (non-vite-ruby) app that can demo sourcemap usage, or if there's anything that could possibly be in vite-ruby that could be affecting this and need fixing?

**Reply by jrochkind** [reply]
Confirmed by someone else that it seems to work in a plain vite demo app at https://github.com/vitejs/vite/discussions/13845#discussioncomment-12074330 , so I think it might be something particular to vite_ruby and the way it invokes source maps in vite?  @ElMassimo would love your thoughts!  

This is the only thing i see in commit history between working and broken versions related to sourceMaps, although it's not clear to me it applies to CSS/SASS source maps at all (instead of just JS). But you might have more insight as to whether it could touch vite_ruby? https://github.com/vitejs/vite/pull/17677

**Reply by ElMassimo** [reply]
There shouldn't be anything in `vite-plugin-ruby` that interferes with source maps, though it could be a Vite bug related to the file paths, given that the Vite root in a Rails project would be `app/frontend` or `app/javascript` instead of the top-level directory.

**Reply by jrochkind** [reply]
Thanks for the replies! Sadly I am still having this problem with vite 5.4.16. 

Downgrade to 5.3.4, I get SASS sourcemaps. Upgrade to vite 5.4.16, with no other changes in my app, I do not get SASS sourcemaps. 

I am still curious if _anyone_ has Sass sourcemaps working with vite-rails and vite 5.4.x? It would be really useful to know if this is easily reproducible by anyone else. (Maybe nobody else uses sass anymore??)

Aha, though -- updating all the way to vite 6.2.4 seems to fix the problem for me.... cross fingers, hope I'm looking at the right thing! 

Having seen in another thread that vite-ruby should work fine with vite 6 and you will make it the default install soon, and it looks like it's working for me -- that's a fine solution!
